internal/backup_list_handler.go

```diff
@@ -17,12 +19,67 @@ func HandleBackupList(folder storage.Folder) {
 		tracelog.ErrorLogger.FatalError(err)
 	}
 
-	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+	WriteBackupList(backups, os.Stdout)
+}
+
+// TODO : unit tests
+func HandleBackupListWithFlags(folder storage.Folder, pretty bool, json bool) {
+	backups, err := getBackups(folder)
+	if err != nil {
+		tracelog.ErrorLogger.FatalError(err)
+	}
+
+	if json {
+		WriteBackupListAsJson(backups, os.Stdout, pretty)
+	} else if pretty {
+		WritePrettyBackupList(backups, os.Stdout)
+	} else {
+		WriteBackupList(backups, os.Stdout)
+	}
+}
+
+// TODO : unit tests
+func WriteBackupList(backups []BackupTime, output io.Writer) {
+	writer := tabwriter.NewWriter(output, 0, 0, 1, ' ', 0)
 	defer writer.Flush()
 	fmt.Fprintln(writer, "name\tlast_modified\twal_segment_backup_start")
-
 	for i := len(backups) - 1; i >= 0; i-- {
 		b := backups[i]
 		fmt.Fprintln(writer, fmt.Sprintf("%v\t%v\t%v", b.BackupName, b.Time.Format(time.RFC3339), b.WalFileName))
 	}
 }
+
+// TODO : unit tests
+func WritePrettyBackupList(backups []BackupTime, output io.Writer) {
+	writer := table.NewWriter()
+	writer.SetOutputMirror(output)
+	defer writer.Render()
+	writer.AppendHeader(table.Row{"#", "Name", "Last modified", "WAL segment backup start"})
+	for i, b := range backups {
+		writer.AppendRow(table.Row{i, b.BackupName, b.Time.Format(time.RFC850), b.WalFileName})
+	}
+}
+
+// TODO : unit tests
+func WriteBackupListAsJson(backups []BackupTime, output io.Writer, pretty bool) {
+	fmt.Print("[")
```

Comment:
Plz, use marshalling to format JSON.
Like json.Marshal(arrayOfCoolThings)

Also, it would be good if this json is pretty-printed, not minified